pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -218,7 +218,8 @@ private void completeConnect() {
             // partitions metadata lookups
             state = State.ProxyLookupRequests;
             lookupProxyHandler = new LookupProxyHandler(service, this);
-            ctx.writeAndFlush(Commands.newConnected(protocolVersionToAdvertise));
+            ctx.writeAndFlush(
+                Commands.newConnected(protocolVersionToAdvertise, service.getConfiguration().getMaxMessagesSize()));
```

Comment:
Proxy should also take into consideration what's the min between proxy and broker.

Eg: if proxy has 5MB and broker has 1MB, proxy should downgrade itself to 1MB

Reply:
Oh, right, proxy is a client of broker so it need receive the max message size from broker and adjust itself. Will fix

Reply:
Yes, the proxy should not even need the config option. Just take the broker value.

Reply:
Yeah, already fixed in the last commit